js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useDefinitionTagFilter.tsx

```diff
@@ -39,27 +42,45 @@ export const useAssetTagFilter = ({
   });
 };
 
+export const useDefinitionTagFilterWithManagedState = ({allTags}: {allTags: Tag[]}) => {
+  const [tags, setTags] = useQueryPersistedState<Tag[]>({
+    encode: (tags) => ({tags}),
+    decode: (qs) => (qs.tags as Tag[]) || [],
+  });
+  return useDefinitionTagFilter({
+    allTags,
+    tags,
+    setTags,
+  });
+};
+
 export const buildDefinitionTag = memoize(
-  (tag: DefinitionTag) => {
+  (tag: Tag) => {
     return tag;
   },
   (tag) => [tag.key, tag.value].join('|@-@|'),
 );
 
-export function useAssetTagsForAssets(
-  assets: {definition?: {tags?: DefinitionTag[] | null} | null}[],
-): DefinitionTag[] {
+export function useTagsForAssets(assets: {definition?: {tags?: Tag[] | null} | null}[]) {
```

Comment:
Only partially related to this line, but I've noticed that sometimes these new resolvers should be non-null and we just forget to use the non_null_list util in python. I'm not sure under what circumstance a definition would be present but tags would be null. Maybe this typing was an error and we can fix it?